k4Gen/src/components/HepMCToEDMConverter.cpp

```diff
@@ -17,6 +17,7 @@ edm4hep::MutableMCParticle HepMCToEDMConverter::convert(std::shared_ptr<const He
   // convert momentum
   auto p = hepmcParticle->momentum();
   edm_particle.setMomentum( {float(p.px()), float(p.py()), float(p.pz())} );
+  edm_particle.setMass( float ( hepmcParticle->generated_mass() ) );
```

Comment:
Mass is `double`
https://github.com/key4hep/EDM4hep/blob/7e1efbeec9f59ca08d57a875c01df36ee2877d31/edm4hep.yaml#L164